src/txmempool.cpp

```diff
@@ -1150,3 +1227,169 @@ size_t CTxMemPool::DynamicMemoryUsage() const {
           cachedInnerUsage);
 }
 
+bool CCoinsViewMemPool::IsQualityValid(const CScCertificate& cert, CAmount certFee) const
+{
+    if (!base->IsQualityValid(cert, certFee))
+    {
+        LogPrint("cert", "%s:%s():%d - cert %s NOK\n", __FILE__, __func__, __LINE__, cert.GetHash().ToString());
+        return false;
+    }
+
+    const uint256& scId = cert.GetScId();
+
+    if ((mempool.mapSidechains.count(scId) != 0) &&
+        (!mempool.mapSidechains.at(scId).mBackwardCertificates.size() == 0))
+    {
+
+        for (const auto& entry : mempool.mapSidechains.at(scId).mBackwardCertificates)
+        {
+            const uint256& memPoolCertHash = entry.second;
+            const CScCertificate& memPoolCert = mempool.mapCertificate.at(memPoolCertHash).GetCertificate();
+
+            // should be nothing but the same epoch
+            if (memPoolCert.epochNumber != cert.epochNumber)
+            {
+                LogPrint("cert", "%s:%s():%d - WARNING: cert %s epoch=%d vs cert %s epoch=%d\n", __FILE__,__func__, __LINE__,
+                    cert.GetHash().ToString(), cert.epochNumber, memPoolCertHash.ToString(), memPoolCert.epochNumber);
+                continue;
+            }
```

Comment:
If this situation happens we have messed up badly the mempool I believe. We should add checks on CTxMempool::check() to make sure this does not happen (or we assert)